src/Features/Core/Portable/ExtractClass/ExtractClassOptions.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.PullMemberUp;
+
+namespace Microsoft.CodeAnalysis.ExtractClass
+{
+    internal class ExtractClassOptions
+    {
+        public ExtractClassOptions(
+            string fileName,
+            string typeName,
+            bool sameFile,
+            ImmutableArray<ExtractClassMemberAnalysisResult> memberAnalysisResults)
+        {
+            FileName = fileName;
+            TypeName = typeName;
+            MemberAnalysisResults = memberAnalysisResults;
+            SameFile = sameFile;
+        }
+
+        public string FileName { get; }
+        public string TypeName { get; }
+        public ImmutableArray<ExtractClassMemberAnalysisResult> MemberAnalysisResults { get; }
+        public bool SameFile { get; }
+    }
+
+    internal class ExtractClassMemberAnalysisResult
+    {
+        /// <summary>
+        /// The member needs to be pulled up.
+        /// </summary>
+        public readonly ISymbol Member;
```

Comment:
can users not select multiple members?  this seems weird.  we have a service already that understands selecting 1-N members at a time.  feels like we should use that here.

Reply:
They can select multiple, but the code refactoring only triggers for one. I'm open to making it do multiple, but neither pull members up nor extract interface works that way. Maybe could be a follow up to make the behavior consistent across all of them?